internal/manager/config.go

```diff
@@ -66,9 +68,10 @@ type Config struct {
 	GatewayAPIControllerName string
 
 	// Ingress status
-	PublishService       string
-	PublishStatusAddress []string
-	UpdateStatus         bool
+	PublishService              string
+	publishServiceNamespaceName types.NamespacedName
```

Comment:
Adding a private field that duplicates the already existing fields seems fishy to me. I'm wondering if we wouldn't like to split this by e.g. leaving the `Config` struct to carry the raw flags that are set by the user (as it is now, no validation whatsoever), and creating a second one (could be named `ValidatedConfig`) that would have a constructor accepting the old `Config`, doing the validations in it. That gives us a clear way to tell that if we accept the `ValidatedConfig` somewhere, we can be sure the data is valid. WDYT?

Reply:
If you find it too much work to tackle it in this PR, I think it still could be worth creating an issue for that - probably there are more cases in which we use some fields from the `Config` and we have to validate them on the go while we could have them checked beforehand.

Reply:
Thanks for the suggestion: added `FlagNamespacedName` to parse and validate command line flags straight into the config field. PTAL

Reply:
Clever! 👍